pyprep/find_noisy_channels.py

```diff
@@ -212,209 +234,229 @@ def find_bad_by_nan_flat(self):
+        chan_amplitudes = _mat_iqr(self.EEGData, axis=1) * IQR_TO_SD
+        amp_sd = _mat_iqr(chan_amplitudes) * IQR_TO_SD
+        amp_median = np.nanmedian(chan_amplitudes)
+
+        # Calculate robust Z-scores for the channel amplitudes
+        amplitude_zscore = np.zeros(self.n_chans_original)
+        amplitude_zscore[self.usable_idx] = (chan_amplitudes - amp_median) / amp_sd
+
+        # Flag channels with amplitudes that deviate excessively from the median
+        abnormal_amplitude = np.abs(amplitude_zscore) > deviation_threshold
+        deviation_channel_mask = np.isnan(amplitude_zscore) | abnormal_amplitude
+
+        # Update names of bad channels by excessive deviation & save additional info
         deviation_channels = self.ch_names_original[deviation_channel_mask]
         self.bad_by_deviation = deviation_channels.tolist()
         self._extra_info["bad_by_deviation"].update(
             {
-                "median_channel_deviation": channel_deviationMedian,
-                "channel_deviation_sd": channel_deviationSD,
-                "robust_channel_deviations": robust_channel_deviation,
+                "median_channel_amplitude": amp_median,
+                "channel_amplitude_sd": amp_sd,
+                "robust_channel_deviations": amplitude_zscore,
             }
         )
 
     def find_bad_by_hfnoise(self, HF_zscore_threshold=5.0):
-        """Determine noise of channel through high frequency ratio.
+        """Detect channels with abnormally high amounts of high-frequency noise.
 
-        Noisiness of the channel is determined by finding the ratio of the median
-        absolute deviation of high frequency to low frequency components.
+        The noisiness of a channel is defined as the amplitude of its
+        high-frequency (>50 Hz) components divided by its overall amplitude.
+        A channel is considered "bad-by-high-frequency-noise" if its noisiness
+        is considerably higher than the median channel noisiness, as determined
+        by a robust Z-scoring method and the given Z-score threshold.
 
-        Low pass 50 Hz filter is used to separate the frequency components.
-        A robust z-score is then calculated relative to all the channels.
+        Due to the Nyquist theorem, this method will only attempt bad channel
+        detection if the sample rate of the given signal is above 100 Hz.
 
         Parameters
         ----------
-        HF_zscore_threshold : float
-            z-score threshold above which channels would be labelled as bad.
+        HF_zscore_threshold : float, optional
+            The minimum noisiness z-score of a channel for it to be considered
+            bad-by-high-frequency-noise. Defaults to 5.0.
 
         """
-        data_tmp = np.transpose(self.EEGData)
-        dimension = np.shape(data_tmp)
-        zscore_HFNoise = np.zeros(self.n_chans_original, dtype=np.float64)
+        MAD_TO_SD = 1.4826  # Scales units of MAD to units of SD, assuming normality
```

Comment:
same as for constant above: could you perhaps even add a link here to why this is the case?